components/Header/Header.tsx

```diff
@@ -0,0 +1,56 @@
+import ToggleTheme from "./ToggleTheme";
+import { Avatar } from "@mui/material";
+import { auth } from "../../firebase";
+import Logo from "../Logo"
+import Link from 'next/link'
+import SearchBar from './SearchBar';
+import SettingsButton from './SettingsButton'
+import AppsButton from './AppsButton';
+import NavLinks from './NavLinks';
+
+const Header = () => {
+  return (
+    <div>
+        {/* Top: Toolbar */}
+        <div 
+        className='grid grid-cols-2 md:grid-cols-3 gap-y-2 grid-flow-row-dense w-full 
+        py-5 px-3 justify-items-stretch items-center bg-white dark:bg-neutralDark-500'>
+            {/* Left: Logo */}
+            <div 
+            className="flex items-center justify-self-start cursor-pointer md:mr-auto
+            text-black dark:text-neutralDark-50">
+                <Link href="/" passHref>
+                    <a><Logo fill="currentColor"/></a>
+                </Link>
+            </div>
+        
+            {/* Center: Search */}
+            <div 
+            className="flex items-center justify-center col-span-2 md:col-span-1 md:justify-items-start">
+                <SearchBar placeholder="What's your question?" />
+            </div>
+        
+            {/* Right: User */}
+            <div 
+            className="flex space-x-2 items-center justify-self-end whitespace-nowrap px-1 md:ml-auto
+            text-neutral-700 dark:text-neutralDark-150">
+                <SettingsButton />
+                <AppsButton />
+                <Avatar
```

Comment:
I wonder if we should make the avatar and mode toggle the same size of the icons (gears and apps). What do you think?

Reply:
Gears and apps are roughly the same size as the theme toggle, do you want me to make the avatar smaller to match the gears?

Alternatively, I could nest the gears and apps in a dropdown that's triggered by clicking on the avatar.

Reply:
whatever looks best. We can leave the icons out. Same size probably looks best. If not, you can leave it as is.